src/kani-driver/src/call_cbmc.rs

```diff
@@ -127,3 +133,192 @@ impl KaniSession {
+        let args: Vec<OsString> =
+            ["kani", "--no-default-checks", "--unwind", "2"].iter().map(|&s| s.into()).collect();
+        let unwind_args = args::KaniArgs::from_iter(args);
+
+        let harnesses = vec![
+            mock_proof_harness("check_one", Some(1)),
+            mock_proof_harness("module::check_two", Some(3)),
+            mock_proof_harness("module::not_check_three", None),
+        ];
+
+        let merged_1 = resolve_unwind_value(
+            &unwind_args,
+            find_proof_harness("check_one", &harnesses).unwrap(),
+        );
+        let merged_2 = resolve_unwind_value(
+            &unwind_args,
+            find_proof_harness("check_two", &harnesses).unwrap(),
+        );
+        let merged_3 = resolve_unwind_value(
+            &unwind_args,
+            find_proof_harness("not_check_three", &harnesses).unwrap(),
+        );
+
+        assert!(merged_1 == Some(1));
+        assert!(merged_2 == Some(3));
+        assert!(merged_3 == Some(2));
+    }
+
+    #[test]
+    fn check_harness_annotation_unwind_resolve() {
+        // Command line unwind value for specific harnesses take precedence over unwind annotation value
+        let args_1: Vec<OsString> =
+            ["kani", "--no-default-checks", "--harness-unwind", "2", "--harness", "check_one"]
+                .iter()
+                .map(|&s| s.into())
+                .collect();
+        let args_2: Vec<OsString> =
+            ["kani", "--no-default-checks", "--harness-unwind", "2", "--harness", "check_two"]
+                .iter()
+                .map(|&s| s.into())
+                .collect();
+        let args_3: Vec<OsString> = [
+            "kani",
+            "--no-default-checks",
+            "--harness-unwind",
+            "2",
+            "--harness",
+            "not_check_three",
+        ]
+        .iter()
+        .map(|&s| s.into())
+        .collect();
+
+        let unwind_args_1 = args::KaniArgs::from_iter(args_1);
+        let unwind_args_2 = args::KaniArgs::from_iter(args_2);
+        let unwind_args_3 = args::KaniArgs::from_iter(args_3);
+
+        let harnesses = vec![
+            mock_proof_harness("check_one", Some(1)),
+            mock_proof_harness("module::check_two", Some(3)),
+            mock_proof_harness("module::not_check_three", None),
+        ];
+
+        let merged_1 = resolve_unwind_value(
+            &unwind_args_1,
+            find_proof_harness("check_one", &harnesses).unwrap(),
+        );
+        let merged_2 = resolve_unwind_value(
+            &unwind_args_2,
+            find_proof_harness("check_two", &harnesses).unwrap(),
+        );
+        let merged_3 = resolve_unwind_value(
+            &unwind_args_3,
+            find_proof_harness("not_check_three", &harnesses).unwrap(),
+        );
+
+        assert!(merged_1 == Some(2));
+        assert!(merged_2 == Some(2));
+        assert!(merged_3 == Some(2));
+    }
+
+    #[test]
+    fn check_default_harness_unwind_resolve() {
```

Comment:
So there's two unwind flags that can be present or not, and an annotation that can be present or not. 

There should be one `#[test]` that has those 8 asserts. And some of those combinations maybe aren't even possible. This is too complex